src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs

```diff
@@ -263,15 +263,23 @@ public async Task<ActionResult> Reset(string id)
                 return Forbid();
             }
 
-            if (!_luceneIndexManager.Exists(id))
-            {
-                return NotFound();
-            }
+            var luceneIndexSettings = await _luceneIndexSettingsService.GetSettingsAsync(id);
 
-            _luceneIndexingService.ResetIndex(id);
-            await _luceneIndexingService.ProcessContentItemsAsync(id);
+            if (luceneIndexSettings != null)
+            {
+                if (!_luceneIndexManager.Exists(id))
+                {
+                    await _luceneIndexingService.CreateIndexAsync(luceneIndexSettings);
+                    await _luceneIndexingService.ProcessContentItemsAsync(id);
+                }
+                else
+                {
+                    _luceneIndexingService.ResetIndex(id);
+                    await _luceneIndexingService.ProcessContentItemsAsync(id);
+                }
```

Comment:
@Skrypt 

Good idea to use the `Activated` event but migrations also use it, so it may happen in a middle of migrations while activating a shell, that may alter tables and create content items, **so i think it is better to do it in a deferred task that will be executed in its own scope, this by using `ShellScope.AddDeferredTask()`** as in `LuceneIndexingContentHandler`.

Also, if content items are created through migrations, a deferred task is already added through the Lucee handler that will execute `IndexingAsync()`. Not sure if it is better that `ProcessContentItemsAsync()` execute before `IndexingAsync()`, if so we would just have to use the `Activating` event instead to enlist the deferred task first. Otherwise for now i have no obvious solution to ensure that the task will be executed last, but my feeling is that the order is not so important.

Reply:
Ok, needs to be tested to see if it actually is the case.

Reply:
Okay no problem

But anyway, better to do it in a deferred task `ShellScope.AddDeferredTask(scope => What you do here)`

Reply:
See #9171